vscode-extension/editor/src/VSCodeEditor.tsx

```diff
@@ -0,0 +1,449 @@
+      let updatedConfig: any = {};
+      try {
+        // Try parsing the string as JSON first
+        updatedConfig = JSON.parse(text);
+      } catch (e) {
+        // If that fails, try parsing the string as YAML
+        updatedConfig = yaml.load(text);
+      }
+
+      setAIConfig(updatedConfig);
+    }
+  }, []);
+
+  // Register an event listener to handle messages from the extension host
+  // This is how we'll receive updates to the webview's content
+  // TODO: saqadri - should this be a useCallback to memoize it?
+  window.addEventListener("message", (event) => {
+    const message = event.data; // The json data that the extension sent
+    if (!message) {
+      console.log("onMessage, MESSAGE=NULL, event=", JSON.stringify(event));
+      return;
+    }
+
+    switch (message.type) {
+      case "update": {
+        console.log("onMessage, message=", JSON.stringify(message));
+        const text = message.text;
+
+        // Update our webview's content
+        updateContent(text);
+        return;
+      }
+      case "set_server_url": {
+        console.log("onMessage, message=", JSON.stringify(message));
+        const url = message.url;
+        setAIConfigServerUrl(url);
+        updateWebviewState(vscode, { serverUrl: url });
+
+        // TODO: saqadri - as soon as content is updated, we have to call
+        // /get endpoint so we get the latest content from the server
+        return;
+      }
+      default: {
+        console.log("onMessage, UNHANDLED message=", JSON.stringify(message));
+        return;
+      }
+    }
+  });
+
+  const loadConfig = useCallback(async () => {
+    const route = ROUTE_TABLE.LOAD(aiConfigServerUrl);
+    const res = await ufetch.post(route, {});
+    // console.log(`IN LOAD: route=${route}, res=${JSON.stringify(res)}`);
+    setAIConfig(res.aiconfig);
+  }, [aiConfigServerUrl]);
+
+  useEffect(() => {
+    if (aiConfigServerUrl !== "") {
+      // This is less important for the first load, but when the webview gets dehydrated and rehydrated,
+      // we'll get the aiConfigServerUrl from the webview state. This will trigger a reload of the config.
+      loadConfig();
+    }
+  }, [aiConfigServerUrl, loadConfig]);
+
+  useEffect(() => {
+    loadConfig();
+  }, [loadConfig]);
+
+  // TODO: saqadri - this should be done extension host-side
+  const setupTelemetryIfAllowed = useCallback(async () => {
+    const isDev = (process.env.NODE_ENV ?? "development") === "development";
+
+    // Don't enable telemetry in dev mode because hot reload will spam the logs.
+    if (isDev) {
+      return;
+    }
+
+    const res = await ufetch.get(
+      ROUTE_TABLE.GET_AICONFIGRC(aiConfigServerUrl),
+      {}
+    );
+
+    const enableTelemetry = res.allow_usage_data_sharing;
+
+    if (enableTelemetry) {
+      datadogLogs.init({
+        clientToken: "pub356987caf022337989e492681d1944a8",
+        env: process.env.NODE_ENV ?? "development",
+        service: "aiconfig-editor",
```

Comment:
nit, we probably want to distinguish vs code from local and gradio here. Maybe "aiconfig-editor-vscode" or "vscode-aiconfig-editor"